server/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorStatus.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.overlord.supervisor;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.common.base.Preconditions;
+
+import java.util.Objects;
+
+/**
+ * This class contains the attributes of a supervisor task which are returned by the API's in
+ * org.apache.druid.indexing.overlord.supervisor.SupervisorResource
+ * and used by org.apache.druid.sql.calcite.schema.SystemSchema.SupervisorsTable
+ */
+@JsonDeserialize(builder = SupervisorStatus.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SupervisorStatus
+{
+  private final String id;
+  private final String state;
+  private final String detailedState;
+  private final boolean healthy;
+  private final SupervisorSpec spec;
+  /**
+   * This is a JSON representation of {@code spec}
+   * The explicit serialization is present here so that users of  {@code SupervisorStatus} which cannot
```

Comment:
Does this work?

Won't the spec still be serialized into `spec`, and won't callers still try to deserialize the `spec` and get some kind of error?

(If it does work -- how does it work? I am perplexed.)